### PR TITLE
Add backend provided slippage

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1357

## What changed (plus any additional context for devs)
- Adds a call to fetch the slippage if the user hasn't defined their own just before executing a swap.

### CODE REVIEW

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 1]
```diff
@@ -554,11 +558,74 @@ export function useSwapInputsController({
   };
 
   const onExecuteSwap = async () => {
-    if (!assetToSell.value || !assetToBuy.value || !quote.value) return;
+    const quoteData = quote.value as Quote | CrosschainQuote;
+    /**
+     * NOTE: Before executing a swap, we want to check several things:
+     * 1. assetToSell && assetToBuy addresses match the quote values
+     * 2. outputNative matches quote value
```
Comment (skylarbarrera) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:565]: this check makes sense but not sure if practical, like isnt there a case where the quote could update

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 2]
```diff
@@ -554,11 +558,74 @@ export function useSwapInputsController({
...
+     */
+    const failedConditions: string[] = [];
+    // Ensure assetToSell and assetToBuy are not null or undefined before proceeding
+    if (!assetToBuy.value || !assetToSell.value) {
+      logger.debug('[onExecuteSwap]: Missing assetToSell or assetToBuy');
+      return;
+    }
+    if ((quote.value as QuoteError)?.error) failedConditions.push('quote error');
+    if (quoteData.buyAmountMinusFees.toString() === '0') failedConditions.push('buyAmountMinusFees is 0');
+    if (quoteData.buyTokenAddress !== assetToBuy.value.address) failedConditions.push('buyTokenAddress mismatch');
```
Comment (skylarbarrera) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:576]: worth checking chains here as well

**Review by skylarbarrera — APPROVED** [review]

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts (hunk 1 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:565]: outputNative is updated when the quote updates

Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:565]: Probably not a necessary check, but is more of a sanity check that the user is displayed and can confirm what will actually get received.

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts (hunk 2 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:576]: knew i was missing something!

**Pull request merged**